tests/games/stendhal/server/maps/quests/FishermansLicenseQuizTest.java

```diff
@@ -0,0 +1,283 @@
+		QuestHelper.setUpBeforeClass();
+		setupZone(ZONE_NAME);		
+	}	
+
+	@Before
+	public void setupBefore() {		
+		// setupQuiz
+		loadQuest(this.quest = new FishermansLicenseQuiz());
+		loadQuest(new FishermansLicenseCollector());		
+
+		npc = getNPC(NPC_NAME);
+		en = npc.getEngine();
+		en.setCurrentState(ConversationStates.ATTENDING);
+	}
+
+	@Override
+	@After
+	public void tearDown() throws Exception {		
+		super.tearDown();
+	}
+
+	@Test
+	public void testHiHelpJob() {		
+		en.setCurrentState(ConversationStates.IDLE);
+		en.stepTest(player, "hi");
+		
+		assertThat(getReply(npc), is("Hello greenhorn!"));
+		
+		en.stepTest(player, "help");
+		assertThat(getReply(npc), is("If you explore Faiumoni you will find several excellent fishing spots."));
+		
+		en.stepTest(player, "job");
+		assertThat(getReply(npc), is("I'm a teacher for fishermen. People come to me to take their #exams."));
+	}
+	
+	@Test
+	public void testQuestWhenExamNotPassed() {
+		en.stepTest(player, "quest");
+		assertEquals(
+				"I don't need anything from you, but if you like, you can do an #exam to get a fisherman's license.",
+				getReply(npc));
+	}
+
+	@Test
+	public void testExamDecline() {
+		
+		en.stepTest(player, "exam");
+		assertThat(getReply(npc), is("Are you ready for the first part of your exam?"));
+		
+		en.stepTest(player, "no");
+		assertThat(getReply(npc), is("Come back when you're ready."));
+		
+		assertLoseKarma(0);
+		assertThat(player.getQuest(FishermansLicenseQuiz.QUEST_SLOT), is("rejected"));		
+	}
+	
+	@Test
+	public void testQuestWhenExamPassedButNotCollector() {
+		player.setQuest(FishermansLicenseQuiz.QUEST_SLOT, "done");
+
+		en.stepTest(player, "quest");
+		assertEquals(
+				"I don't need anything from you, but if you like, you can do an #exam to get a fisherman's license.",
+				getReply(npc));
+	}	
+
+	@Test
+	public void testExamFailure() {
+		
+		player.setQuest(FishermansLicenseQuiz.QUEST_SLOT, null);
+		
+		Set<String> fishInExam = ImmutableSet.of("trout", "perch",
+				"mackerel", "cod", "roach", "char", "clownfish", "surgeonfish");
+		
+		en.stepTest(player, "exam");
+
+		assertThat(getReply(npc), is("Are you ready for the first part of your exam?"));
+		assertThat(en.getCurrentState(), is(ConversationStates.QUEST_OFFERED));
+
+		en.stepTest(player, "yes");
+
+		assertThat(getReply(npc), is("Fine. The first question is: What kind of fish is this?"));
+
+		System.out.println("TEST zone " + zone.getName() + " " + zone.hashCode());
+		
+		
+		// assert fish on the table
+		Optional<Item> itemOnGround = zone.getItemsOnGround().stream().findFirst();
+		
+		assertThat(itemOnGround.isPresent(), is(true));
+		assertThat(itemOnGround.get().getItemClass(), is("food"));
+		
+		assertThat(fishInExam, hasItem(itemOnGround.get().getName()));
+		
+		// wrong answear
+		en.stepTest(player, "notAFish");
+		assertThat(getReply(npc), is("No, that's wrong. Unfortunately you have failed, but you can try again tomorrow."));

```

Comment:
```suggestion
		assertThat(getReply(npc), is("No, that's wrong. Unfortunately, you have failed, but you can try again tomorrow."));

```

Reply:
The quest text if "Unfortunately you have failed, but you can try again tomorrow", so adding the comma would make it fail.